packages/account/src/providers/provider.ts

```diff
@@ -955,40 +957,62 @@ export default class Provider {
    * @returns A promise that resolves to the estimated transaction request object.
    */
   async estimatePredicates<T extends TransactionRequest>(transactionRequest: T): Promise<T> {
-    const shouldEstimatePredicates = Boolean(
-      transactionRequest.inputs.find(
-        (input) =>
-          'predicate' in input &&
-          input.predicate &&
-          !equalBytes(arrayify(input.predicate), arrayify('0x')) &&
-          new BN(input.predicateGasUsed).isZero()
-      )
+    const shouldEstimatePredicates = transactionRequest.inputs.some(
+      (input) => isPredicate(input) && bn(input.predicateGasUsed).isZero()
     );
+
     if (!shouldEstimatePredicates) {
       return transactionRequest;
     }
+
     const encodedTransaction = hexlify(transactionRequest.toTransactionBytes());
+
     const response = await this.operations.estimatePredicates({
       encodedTransaction,
     });
 
-    const {
-      estimatePredicates: { inputs },
-    } = response;
+    const { estimatePredicates } = response;
 
-    if (inputs) {
-      inputs.forEach((input, index) => {
-        if ('predicateGasUsed' in input && bn(input.predicateGasUsed).gt(0)) {
-          // eslint-disable-next-line no-param-reassign
-          (<CoinTransactionRequestInput>transactionRequest.inputs[index]).predicateGasUsed =
-            input.predicateGasUsed;
-        }
-      });
-    }
+    // eslint-disable-next-line no-param-reassign
+    transactionRequest = this.parseEstimatePredicatesResponse(
+      transactionRequest,
+      estimatePredicates
+    );
 
     return transactionRequest;
   }
 
+  async estimatePredicatesAndGasPrice<T extends TransactionRequest>(
```

Comment:
Should we have some API docs here?

Reply:
https://github.com/FuelLabs/fuels-ts/pull/3676/commits/1d6510db8da38d1642e585badd5e0930d7a89094

Reply:
Btw, did you think about packing `estimateGasPrice` with the `GetChain` as well?

Reply:
@arboleya Apologies, I misread your question earlier.

Currently, we don’t have any documentation covering `estimatePredicates` or `estimateGasPrice`.

If we decide to add documentation for these, I believe they should be included in detailed guide on manually submitting a transaction. Or even add these details to the existent guides that we have [here](https://docs.fuel.network/docs/fuels-ts/transactions/modifying-the-request/)

Should we add a new issue for this?

Reply:
Got it. I think it makes sense.

Reply:
@arboleya https://github.com/FuelLabs/fuels-ts/issues/3677